Formula/dlv.rb

```diff
@@ -0,0 +1,17 @@
+class Dlv < Formula
+  desc "Debugger for the Go programming language"
+  homepage "https://github.com/go-delve/delve"
+  url "https://github.com/go-delve/delve/archive/v1.6.0.tar.gz"
+  sha256 "a10a6fc40d87572c6d3f3becdb1a289269e17526d038749f2fa04dd9f591f26a"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "./cmd/dlv"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/dlv version").chomp
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I add a reasonable test to this formula.
Sometimes it is hard to design a proper test and some of the existing formulae give bad examples for this.